glotpress.php

```diff
@@ -164,12 +164,28 @@ function gp_unsupported_permalink_structure_admin_notice() {
 
 require_once GP_PATH . 'gp-settings.php';
 
+/*
+ * Check if we need to run the upgrade routine, but only run it on the admin side.
+ */
+if ( is_admin() && GP_DB_VERSION > get_option( 'gp_db_version' ) ) {
+	require_once GP_PATH . GP_INC . 'install-upgrade.php';
+	gp_upgrade_db();
+}
+
```

Comment:
Just above this code, we include the `gp-settings.php` file, where we have this code. Do you need to move it here, or maybe it can stay in the previous place?

Reply:
In https://github.com/GlotPress/GlotPress/pull/1764/commits/44b6ceefcc80d37a454bea20a856b158b30794e7, I've moved the code from `gp-settings.php` to here.

My reasoning for moving it was that installing/upgrading the database feels like something that would live in the main `glopress.php` file, as that's related to the "lifecycle" of the plugin. It also seemed appropriate since we are now also doing the db install/upgrade on plugin activation, with that code also being in `glotpress.php`.

It felt appropriate that all db install/upgrade code would be in `glotpress.php`, instead of both in `glotpress.php` and `gp-settings.php`.

Does this make sense? Let me know if you want me to change this, I'm happy to do so if you don't agree.

Reply:
Something I forgot to mention:

The `gp_upgrade_db()` is called in two places now, so if the function doesn't do the requires, they need to be added twice, before each call to the function.

Alternatively, they could be added to the top level of `glotpress.php`, but that would mean that they would be imported even if there is no need to upgrade the db.